proposals/NNNN-package-access-modifier.md

```diff
@@ -0,0 +1,326 @@
+
+### Package Symbols and `@inlinable` Functions
+
+`package` functions can be made `@inlinable`.  Just like with `@inlinable public`, not all symbols are usable within the function: they must be `open`, `public`, `package`, or `@usableFromInline`.  Note that `@usableFromInline` allows the use of a symbol from `@inlinable` functions whether they're `package` or `public`.  `@usableFromPackageInline` is introduced to export a symbol only for use by `@inlinable package` functions.
+
+`package` functions can be made `@inlinable`.  Just like with `@inlinable public`, not all symbols are usable within the `@inlinable package` function: they must be `open`, `public`, `package`, `@usableFromInline`, or `usableFromPackageInline`.
+`@usableFromPackageInline` is a new attribute which allows a symbol to be used from `@inlinable package` functions (that are defined in the same module) without having to make the symbol `package` or `public`.  It can be used anywhere that `@usableFromInline` can be used, but the attributes cannot be combined.  A `@usableFromPackageInline` symbol must be `internal`. For example: 
+
+```
+@usableFromPackageInline func internalFuncA() {}
+
+@inlinable package func pkgUse() {
+    internalFuncA() // OK
+}
+
+@inlinable public func publicUse() {
+    internalFuncA() // OK
+}
+```
+
+The existing `@usableFromInline` attribute can be applied to `package` symbols as well as `internal` symbols.  This allows those symbols to be used from `@inlinable public` functions (that are defined anywhere in the same package) without having to make them `public`.  The Swift frontend will include `@usableFromInline package` symbols in the ordinary `.swiftinterface` for a module, not its `.package.swiftinterface`.
+
+```
+@usableFromPackageInline func internalFuncA() {}
+@usableFromInline func internalFuncB() {}
+@usableFromInline package func packageFunc() {}
+
+@inlinable package func pkgUse() {
+    internalFuncA() // OK
+    internalFuncB() // OK
+    packageFunc() // OK
+}
+
+@inlinable public func publicUse() {
+    internalFuncA() // ERROR
+    internalFuncB() // OK
+    packageFunc() // OK
+}
+```
+ 
+### Resiliency
+
+Library evolution makes modules resilient.  We can incorporate a package name into a resiliency check and bypass it if modules are in the same package.  This will remove the need for resilience overhead such as indirection and language requirements such as `@unknown default` for an unfrozen `enum`.  
```

Comment:
I'm sorry for overlooking this section in earlier passes.  The "we can" makes it unclear whether you're actually proposing this as part of this proposal.  If you're proposing it, you should say so affirmatively, like "The Swift frontend will assume that modules defined in the same package will always be rebuilt together and do not require a resilient ABI boundary between them.  This removes the need for performance and code size overhead introduced by ABI-resilient code generation, and it also eliminates language requirements such as `@unknown default` for a non-`frozen` `enum`."  However, I suspect you mean this to just be a future direction, right?